File: tests/PicoJsonRpc.Tests/Fixtures/echo_server.mjs
// NDJSON echo fixture — mirrors the A0 string-embedded wire contract: reads
// request frames on stdin, replies a response frame per line with the request
// id preserved and the request echoed as raw JSON text in `result`.
import readline from "node:readline";
const rl = readline.createInterface({ input: process.stdin, terminal: false });
rl.on("line", (line) => {
  try {
    const req = JSON.parse(line);
    const out = { jsonrpc: "2.0", id: req.id ?? null };
    out.result = JSON.stringify({ echo: req });
    process.stdout.write(JSON.stringify(out) + "\n");
  } catch {
    process.stdout.write(
      JSON.stringify({ jsonrpc: "2.0", id: null, error: { code: -32700, message: "parse" } }) + "\n"
    );
  }
});
process.stdin.on("end", () => process.exit(0));

File: tests/PicoJsonRpc.Tests/Fixtures/echo_server.py
# NDJSON echo fixture (Python parity of echo_server.mjs) — reads request
# frames on stdin, replies a response frame per line with the request id
# preserved and the request echoed as raw JSON text in `result`.
import json
import sys

for line in sys.stdin:
    line = line.strip()
    if not line:
        continue
    try:
        req = json.loads(line)
        out = {"jsonrpc": "2.0", "id": req.get("id")}
        out["result"] = json.dumps({"echo": req})
        sys.stdout.write(json.dumps(out) + "\n")
    except Exception:
        sys.stdout.write(
            json.dumps({"jsonrpc": "2.0", "id": None, "error": {"code": -32700, "message": "parse"}}) + "\n"
        )
    sys.stdout.flush()